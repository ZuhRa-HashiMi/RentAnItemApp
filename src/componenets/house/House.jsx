import React from 'react'
import './house.css'
import { useDispatch } from 'react-redux'
import { deleteHouse } from '../../redux/house/house'

const House = (props) => {
  const {data, showBtnDelete} = props
  const {id, picture, name, price} = data

  const dispatch = useDispatch();
  const handleRemove = () => {
    dispatch(deleteHouse(id));
  };
  return (
    <>
      <div className='column house'>
        <img src={picture} alt={`house ${name}`} />
        <div className='house-description'>
          <div className='row space-between'>
            <h4 className='max-content-flex'>{name}</h4>
            <h4>$ {price}</h4>
          </div>
          <div className='row space-between'>
            <div></div>
            <p>per Month</p>
          </div>
        </div>
        {
          showBtnDelete &&
          <button className='max-content-flex btn-default'
          onClick={handleRemove}>Delete</button>
        }
      </div>
    </>
  )
}

export default House