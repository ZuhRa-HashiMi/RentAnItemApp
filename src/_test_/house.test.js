import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import React from 'react';
import { testListHouses } from './setupTests';
import House from '../componenets/house/House';

describe('test render', () => {
  test('House should match snapshoot', () => {
    const tree = renderer.create(
      <React.StrictMode>
        <House data={testListHouses[0]} />
      </React.StrictMode>,
    ).toJSON();

    expect(tree).toMatchSnapshot()
  })
})
