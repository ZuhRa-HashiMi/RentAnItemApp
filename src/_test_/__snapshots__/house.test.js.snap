// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render House should match snapshoot 1`] = `
<div
  className="column house"
>
  <img
    alt="house Pent house"
    src="https://github.com/shella12/RentAnItemBackend/assets/44798044/5bb4ea2c-4ae5-45b8-9083-aeafbfbcf287"
  />
  <div
    className="column house-content-wrapper"
  >
    <div
      className="column max-content-flex house-content"
    >
      <div
        className="row space-between"
      >
        <h4
          className="max-content-flex house-title"
        >
          Pent house
        </h4>
        <h4>
          $ 
          300000.0
           
          <br />
          per Month
        </h4>
      </div>
      <p
        className="desktop-only house-description"
      >
        This house has 5 rooms upstairs and 4 rooms downstairs. Home design emphasizes clean lines and geometric shapes.
         The magic thing about this houe is that it feels good to leave, and it feels even better to come back.
         Modern Conveniences and Green Features. Well-Placed Bathrooms. Quality Construction with Distinct Architecture. 
         Plenty of Storage Space. It's important to have designated storage areas and large closets.
      </p>
    </div>
  </div>
</div>
`;
